src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/ValidateAnomalyDetectorActionHandler.java

```diff
@@ -0,0 +1,590 @@
+            String errorMsg = feature.getName() + ": feature query is potentially wrong as no hits were found";
+            suggestedChangesMap.computeIfAbsent("feature_attributes", k -> new ArrayList<>()).add(errorMsg);
+        }
+        if (counter.get() == anomalyDetector.getFeatureAttributes().size()) {
+            if (!suggestedChangesMap.isEmpty()) {
+                validateAnomalyDetectorResponse();
+            } else {
+                intervalValidation();
+            }
+        }
+    }
+
+    private long[][] createNewTimeRange(long detectorInterval) {
+        long timeRanges[][] = new long[MAX_NUM_OF_SAMPLES_VIEWED][2];
+        long delayMillis = Optional
+            .ofNullable((IntervalTimeConfiguration) anomalyDetector.getWindowDelay())
+            .map(t -> t.toDuration().toMillis())
+            .orElse(0L);
+        long dataEndTime = Instant.now().toEpochMilli() - delayMillis;
+        long dataStartTime = dataEndTime - ((long) (MAX_NUM_OF_SAMPLES_VIEWED) * detectorInterval);
+        for (int j = 0; j < MAX_NUM_OF_SAMPLES_VIEWED; j++) {
+            timeRanges[j][0] = dataStartTime + (j * detectorInterval);
+            timeRanges[j][1] = timeRanges[j][0] + detectorInterval;
+        }
+        return timeRanges;
+    }
+
+    private synchronized void intervalValidation() throws IOException {
+        long detectorInterval = Optional
+            .ofNullable((IntervalTimeConfiguration) anomalyDetector.getDetectionInterval())
+            .map(t -> t.toDuration().toMillis())
+            .orElse(0L);
+        for (long i = detectorInterval; i <= MAX_INTERVAL_LENGTH; i *= INTERVAL_RECOMMENDATION_MULTIPLIER) {
+            long timeRanges[][] = createNewTimeRange(i);
+            // Need to try and check if the infering logic is done before calling on the method again
+            // with a new interval since otherwise the requests get mixed up
+            try {
+                if (inferAgain.get()) {
+                    samplingHelper(timeRanges, i);
+                }
+                wait();
+            } catch (Exception ex) {
+                onFailure(ex);
+            }
+        }
+    }
+
+    private List<String> getFeatureFieldNames() throws IOException {
+        List<String> featureFields = new ArrayList<>();
+        for (Feature feature : anomalyDetector.getFeatureAttributes()) {
+            ParseUtils.parseAggregators(feature.getAggregation().toString(), xContent, feature.getId());
+            XContentParser parser = XContentType.JSON
+                .xContent()
+                .createParser(xContent, LoggingDeprecationHandler.INSTANCE, feature.getAggregation().toString());
+            parser.nextToken();
+            List<String> fieldNames = parseAggregationRequest(parser);
+            featureFields.add(fieldNames.get(0));
+        }
+        return featureFields;
+    }
+
+    private synchronized void samplingHelper(long[][] timeRanges, long detectorInterval) throws IOException {
```

Comment:
name it `verifyWithInterval()` or more specific name

Reply:
done